consensus/scp/tests/mock_network/mod.rs

```diff
@@ -379,24 +302,25 @@ impl SCPNode {
 
         let thread_shared_data = Arc::clone(&node.shared_data);
         let thread_local_node = Arc::clone(&node.local_node);
+        let mut allow_propose: bool = true;
         let mut current_slot: usize = 0;
         let mut total_broadcasts: u32 = 0;
 
         let thread_handle = Some(
             thread::Builder::new()
                 .name(thread_name)
                 .spawn(move || {
-
                     // All values that have not yet been externalized.
                     let mut pending_values: HashSet<String> = HashSet::default();
 
                     'main_loop: loop {
                         // Collect and process any messages we have received
                         let mut incoming_msgs = Vec::<Arc<Msg<String>>>::new();
 
-                        for msg in receiver.try_iter() {
-                            // Handle message based on it's type
-                            match msg {
+                        // Handle one incoming message based on it's type
+                        match receiver.try_recv() {
```

Comment:
A side effect of this change is that the time to stop the thread would be the number of messages in queue times the time it takes to process a message. I am not sure if that is an issue or not.

Reply:
I agree with you but I don't think it matters... or at least I haven't seen any issues with shutting threads down...